kedro/framework/cli/starters.py

```diff
@@ -487,71 +487,66 @@ def _get_extra_context(  # noqa: PLR0913
             None in case the flag wasn't used.
 
     Returns:
-        the prompts_required dictionary, with all the redundant information removed.
+        Config dictionary, passed the necessary processing, with default values if needed.
     """
+    # Obtain config from file or user prompts, validate and provide default values
     if config_path:
-        extra_context = _fetch_config_from_file(config_path)
-        _validate_config_file_against_prompts(extra_context, prompts_required)
-        _validate_config_file_inputs(extra_context, starter_alias)
+        extra_context = _fetch_validate_parse_config_from_file(
+            config_path, prompts_required, starter_alias
+        )
     else:
-        extra_context = _fetch_config_from_user_prompts(
+        extra_context = _fetch_validate_parse_config_from_user_prompts(
             prompts_required, cookiecutter_context
         )
 
-    # Format
-    extra_context.setdefault("kedro_version", version)
-
-    tools = _convert_tool_names_to_numbers(selected_tools)
-
-    if tools is not None:
-        extra_context["tools"] = tools
-
+    # Update extra_context, if CLI inputs are available
+    if selected_tools is not None:
+        extra_context["tools"] = _convert_tool_names_to_numbers(selected_tools)  # type: ignore
```

Comment:
I've put a lot of effort into making these `type: ignore` statements specific (see https://github.com/kedro-org/kedro/pull/3490) so please update that here as well. I'm seeing quite a lot though, so I'm wondering if there's anything that can be done to not just ignore but actually fix the issues?

Reply:
Sounds good, I'll hold off until you merge #3490. After that, I'll update my code to align with our new --strict standards.

Reply:
fixed all type: ignore